kv/txn_coord_sender.go

```diff
@@ -769,6 +769,8 @@ func (tc *TxnCoordSender) updateState(ctx context.Context, ba roachpb.BatchReque
 				// continue its transaction after receiving ConditionFailedError, which
 				// can come from a unique index violation.
 			}
+			newTxn.Update(pErr.GetTxn())
+			pErr.SetTxn(newTxn)
```

Comment:
Sure, fixed. We now update `newTxn` on `roachpb.TransactionStatusError` as well.

Tweaked `TestTxnCoordSenderTxnUpdatedOnError` as we expect `pErr.GetTxn()` returns a non-nil for errors used in the test.